test/cluster/mzcompose.py

```diff
@@ -261,6 +262,49 @@ def extract_frontiers(output: str) -> Tuple[str, str]:
     assert t_upper, "t has empty upper frontier"
 
 
+def workflow_test_github_15799(c: Composition) -> None:
+    """
+    Test that querying arranged introspection sources on a replica does not
+    crash other replicas in the same cluster that have introspection disabled.
+
+    Regression test for https://github.com/MaterializeInc/materialize/issues/15799.
+    """
+
+    c.down(destroy_volumes=True)
+    c.up("materialized")
+    c.up("computed_1")
+    c.up("computed_2")
+    c.wait_for_materialized()
+
+    c.sql(
+        """
+        CREATE CLUSTER cluster1 REPLICAS (
+            logging_on (
+                REMOTE ['computed_1:2100'],
+                COMPUTE ['computed_1:2102'],
+                WORKERS 2
+            ),
+            logging_off (
+                REMOTE ['computed_2:2100'],
+                COMPUTE ['computed_2:2102'],
+                WORKERS 2,
+                INTROSPECTION INTERVAL 0
+            )
+        );
+        SET cluster = cluster1;
+        SET cluster_replica = logging_on;
+        SELECT * FROM mz_internal.mz_active_peeks, mz_internal.mz_compute_exports;
+        """
+    )
+
+    print("Sleeping to wait for computed_2 to crash")
+    time.sleep(1)
```

Comment:
Hm, not a fan of sleep in CI. How about we turn off `computed_1` and do another SELECT ? Then `computed_2` is forced to reply and the test would time out if `computed_2` crashed?

Reply:
We can also just use replica targeting to query computed_2! I didn't realize that it is not restarted automatically in this test environment.